cranelift/codegen/src/opts/algebraic.isle

```diff
@@ -537,3 +537,20 @@
 ;; both of them since they cancel out.
 (rule (simplify (fmul ty (fneg ty x) (fneg ty y)))
       (fmul ty x y))
+
+
+;; Wasmtime loves doing load+bitcast. This prevents the x64 backend from
+;; merging the load into the op. Remove the bitcasts if possible.
+(rule (simplify (bitcast cty cflags (load lty lflags addr offset)))
+      ;; Two preconditions for this rule:
+      ;; 1. The types must have the same size. (this is enforced by the verifier)
+      ;; 2. The endiannes on both memflags must match
+      (if-let $true (mem_flags_endianness_eq cflags lflags))
```

Comment:
I've been sitting here trying to wrap my head around endianness and lane casts and how it affects s390x and I'll admit I still don't fully grok the implications of this. Natively it seems like having the same endianness is what's required here, but I might also natively think that "well just take the bitcast's endianness and use that on the load" although I'm not certin that's right.

To help assist with assurances here, though, could you also add some runtests which trigger this optimization and work before/after your PR?